source/common/json/config_schemas.cc

```diff
@@ -448,6 +448,7 @@ const std::string Json::Schema::ROUTE_ENTRY_CONFIGURATION_SCHEMA(R"EOF(
       "prefix" : {"type" : "string"},
       "path" : {"type" : "string"},
       "cluster" : {"type" : "string"},
+      "cluster_header" : {"type" : "string"},
       "weighted_clusters": {"$ref" : "#/definitions/weighted_clusters"},
       "host_redirect" : {"type" : "string"},
```

Comment:
I suggest a one of here. cluster,cluster_header,weighted_cluster. That allows you to get rid of the checks in config_impl.cc